### PR TITLE
Logging for Splunk with SLF4J

### PR DESCRIPTION
**Pull request by pcunninghamatwombatsecurity**

### CODE REVIEW

File: _posts/2017-07-14-logging-for-splunk-with-slf4j.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	   <artifactId>logstash-logback-encoder</artifactId>
+	   <version>4.11</version>
+	 </dependency>
+
+### Configuring Logback JSON Encoder
+
+After trying a few different configurations, I settled on a configuration using the `net.logstash.logback.encoder.LoggingEventCompositeJsonEncoder` as the primary encoder for Logback.
+
+Here is the basic configuration I used for testing:
+
```
Comment (laurawhalin) [inline _posts/2017-07-14-logging-for-splunk-with-slf4j.md]: Can you wrap all your code and outputs with `{% highlight plaintext %}` and `{% endhighlight %}`? Thanks!

File: _posts/2017-07-14-logging-for-splunk-with-slf4j.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	}
+	...
+
+Because Logstash now sees these arguments as StructuredArguments, it knows to add them as key/values in the JSON. The output looks like this:
+
+	{"@timestamp":"2017-06-13T20:35:45.881+00:00","classname":"SplunkingAgent","method":"testArguments","file":"SplunkingAgent.java","line":112,"thread":"main","logger_name":"TestArguments","level":"INFO","message":"Things happened: foo=FOO, bar=BAR","foo":"FOO","bar":"BAR","bang":"BANG"}
+
+You can see that the first two arguments, `foo` and `bar`, are output in the message format as `Things happened: foo=FOO, bar=BAR`, but the value for `bang` is not used. However all three values are represented as separate JSON key/values: `"foo":"FOO","bar":"BAR","bang":"BANG"`
+
+A convenient side-effect is that the ordering of the arguments is less important becaue the key names don't need to be ordered in the message format.
```
Comment (laurawhalin) [inline _posts/2017-07-14-logging-for-splunk-with-slf4j.md]: because is missing the 's'

File: _posts/2017-07-14-logging-for-splunk-with-slf4j.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+author_github:  pcunninghamatwombatsecurity
+date: 2017-07-14 06:00:00
+comments: true
+---
+
+
+*Creating JSON Formatted Logs in Java*
+
+If you are using [Splunk for logfile analysis](https://www.splunk.com/), you should also be using a properly formatted output format that allows for the reliable transmission of key/value pairs from your application. JSON is perfect for this and [Logstash](https://www.elastic.co/products/logstash) has created a convenient Logback encoder for us to use. This blog will show you how to
+
```
Comment (laurawhalin) [inline _posts/2017-07-14-logging-for-splunk-with-slf4j.md]: I like to add `<!--more-->` after the first paragraph when there is a single-line opening like on line 12. That way, on the post index page, we'll get a better preview of this post.

**Comment by pcunninghamatwombatsecurity** [comment]
Closing this pull request. Opening an updated one.

**Pull request closed without merging**